tools/ctl/ide/benchmarks/benchmarks.go

```diff
@@ -0,0 +1,49 @@
+package benchmarks
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/util"
+	"github.com/google/oss-rebuild/tools/benchmark"
+	"github.com/pkg/errors"
+)
+
+type Repository interface {
+	List() ([]string, error)
+	Load(string) (*benchmark.PackageSet, error)
+}
+
+type fsRepository struct {
+	fs billy.Filesystem
+}
+
+var _ Repository = (*fsRepository)(nil)
+
+func NewFSRepository(fs billy.Filesystem) *fsRepository {
+	return &fsRepository{fs: fs}
+}
+
+func (r *fsRepository) List() ([]string, error) {
+	all := []string{}
+	err := util.Walk(r.fs, ".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			// Best effort reading, skip failures.
+			return nil
+		}
+		if filepath.Ext(path) != ".json" {
+			return nil
+		}
+		all = append(all, path)
+		return nil
+	})
+	if err != nil {
```

Comment:
same comment as in other PR: if you suppress all the errors in the Walk callback, Walk can't return an err. I'd remove one or the other handlers